FEAT: xgboost handler

isabelizimm opened the pull request:
closes #27 

TODO: 
- [X] research serialization of `xgboost.DMatrix` for `predict`
⬆️ Users are able to post a pd.DataFrame for vetiver to do the handling into an `xgboost.DMatrix` for prediction.

machow reviewed (commented):
The code LGTM w/ Hassan's comments! I wonder if there's a good place to let people find handlers from the docs? (either a written list of which packages are supported, or a list of handlers?)

<img width="367" alt="image" src="https://user-images.githubusercontent.com/2574498/186914129-5339b95f-e1a6-46dd-a95f-b96118aa36f4.png">

isabelizimm commented:
> I wonder if there's a good place to let people find handlers from the docs? (either a written list of which packages are supported, or a list of handlers?)

Good point! I added another tab "Model Types" and updated the front page to reflect what is supported

isabelizimm commented on vetiver/handlers/xgboost.py:
```diff
@@ -0,0 +1,75 @@
+            Prediction from model
+        """
+
+        if xgb_exists:
```
:O this is great, updating all the handlers!
